src/wrapper.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const Promise = require('bluebird');
+const SecretStore = require('./lib/secretstore');
+const KMSFLAG = require('./kmsflag');
+
+class Wrapper {
+    /**
+     *
+     * @param KMSOptions KMS Option
+     */
+    constructor(KMSOptions = {
+        region: null,
+        table: null,
+        kmsKey: 'alias/yadda-secrets',
+    }){
+        this.options = KMSOptions;
+
+        /**
+         * Secret Storage
+         * @type {SecretStore|null}
+         */
+        this.store = null;
+    }
+
+    /**
+     * Get Secret Storage
+     * @return {SecretStore}
+     */
+    get storage(){
+        if(this.store)
+            return this.store;
+
+        const { region, table, kmsKey } = this.options;
+        if(region && table)
+            this.store = new SecretStore({ table, awsOpts: { region }, kmsKey });
+
+        return this.store;
+
+        throw new Error(`Region '${region}' or table '${table}' is invalid`);
```

Comment:
Will this ever throw, seems unreachable

Reply:
Good catch, originally it was throwable.